deployment/src/main/java/io/quarkiverse/rabbitmqclient/deployment/QuarkusRabbitmqClientProcessor.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkiverse.rabbitmqclient.deployment;
+
+import io.quarkiverse.rabbitmqclient.RabbitMQClientBuildConfig;
+import io.quarkiverse.rabbitmqclient.RabbitMQClientProducer;
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
+
+class QuarkusRabbitmqClientProcessor {
+
+    private static final String FEATURE = "rabbitmq-client";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void addDependencies(BuildProducer<IndexDependencyBuildItem> indexDependency) {
+        indexDependency.produce(new IndexDependencyBuildItem("com.rabbitmq", "amqp-client"));
+    }
+
```

Comment:
Why do you index the full `amqp-client` ?
As you use a a producer bean this should not be needed.

Reply:
Thanks for the feedback @loicmathieu. Creating extensions is pretty new to me and I don't know all the ins and outs of everything yet. I look a lot at other extensions and at the documentation. I'll fix all issues you pointed out.

Reply:
Removed